ext/winevt/winevt_utils.cpp

```diff
@@ -572,11 +572,24 @@ render_system_event(EVT_HANDLE hEvent)
   }
 
   EventID = pRenderedValues[EvtSystemEventID].UInt16Val;
-  if (EvtVarTypeNull != pRenderedValues[EvtSystemQualifiers].Type) {
-    EventID = MAKELONG(pRenderedValues[EvtSystemEventID].UInt16Val,
-                       pRenderedValues[EvtSystemQualifiers].UInt16Val);
+  // Default condition does not preserve qualifiers key.
+  if (!does_preserve_qualifiers) {
+    if (EvtVarTypeNull != pRenderedValues[EvtSystemQualifiers].Type) {
+      EventID = MAKELONG(pRenderedValues[EvtSystemEventID].UInt16Val,
+                         pRenderedValues[EvtSystemQualifiers].UInt16Val);
+    }
+
+    rb_hash_aset(hash, rb_str_new2("EventID"), ULONG2NUM(EventID));
+  } else {
```

Comment:
`!does_preserve_qualifiers` && `else` is confusing since it's double negative.
I prefer inverting the condition :

```c
if (preserve_qualifiers) {
  ...
} else {
  ...
}
```

Reply:
Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/25/commits/52ad016c0abd43adc54cd0677b281a178649f6aa